osbs/build/user_params.py

```diff
@@ -98,17 +98,67 @@ def value(self, val):  # pylint: disable=W0221
 
 
 class BuildCommon(object):
-    def __init__(self):
+    def __init__(self, build_json_dir=None):
+        self.arrangement_version = BuildParam("arrangement_version", allow_none=True)
+        self.build_json_dir = BuildParam('build_json_dir', default=build_json_dir)
+        self.component = BuildParam('component')
+        self.filesystem_koji_task_id = BuildParam("filesystem_koji_task_id", allow_none=True)
         self.image_tag = BuildParam("image_tag")
         self.koji_target = BuildParam("koji_target", allow_none=True)
+        self.koji_task_id = BuildParam('koji_task_id', allow_none=True)
         self.platform = BuildParam("platform", allow_none=True)
-        self.arrangement_version = BuildParam("arrangement_version", allow_none=True)
-        self.filesystem_koji_task_id = BuildParam("filesystem_koji_task_id", allow_none=True)
+        self.orchestrator_deadline = BuildParam('orchestrator_deadline', allow_none=True)
+        self.scratch = BuildParam('scratch', allow_none=True)
         self.user = UserParam()
-        self.component = BuildParam('component')
+        self.worker_deadline = BuildParam('worker_deadline', allow_none=True)
+
         self.required_params = [
+            self.build_json_dir,
             self.koji_target,
+            self.user,
         ]
+        self.convert_dict = {}
+
+        # Defaults
+        self.arrangement_version.value = REACTOR_CONFIG_ARRANGEMENT_VERSION
+
+    def attrs_finalizer(self):
```

Comment:
It cannot be cleared because it is used for creating json output.
I can call it only once in sub-classes, otherwise it would be raising errors about duplication.

Better way would be to utilize metaprogramming and do checks about name uniqueness during class creation, not during object initialization (that's too late). Also `required` params should be defined by `required=True|Fase` kwarg instead of specifying an additional attribute for that.

But this change would be quite invasive I can open a tech debt issue